File: Data Structures & Algorithms/car-fleet/submission-3.js
class Solution {
    /**
     * @param {number} target
     * @param {number[]} position
     * @param {number[]} speed
     * @return {number}
     */
    carFleet(target, position, speed) {
        const pair = position.map((p, i) => [p, speed[i]]);
        pair.sort((a, b) => b[0] - a[0]);
        const stack = [];
        for (const [p, s] of pair) {
            stack.push((target - p) / s);
            if (stack.length >= 2 && stack.at(-1) <= stack.at(-2)) {
                stack.pop();
            }
        }
        return stack.length;
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        close_to_open = {
            ")": "(",
            "}": "{",
            "]": "[",
        }

        stack = []
        for c in s:
            if c in close_to_open:
                if len(stack) > 0 and close_to_open[c] == stack[-1]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)
        return len(stack) == 0